src/Backend/App.vala

```diff
@@ -159,16 +155,18 @@ public class Slingshot.Backend.App : Object {
                 count_visible = prop_value.get_boolean ();
             }
         }
-
-        unity_update_info ();
     }
 
     public void unity_reset () {
         unity_sender_name = null;
         count_visible = false;
         current_count = 0;
+    }
 
-        unity_update_info ();
+    public void remove_launcher_entry (string sender_name) {
```

Comment:
Both ways are really similar, I just prefer this way so that the `unity_sender_name` property is private as it should